Students/tylerkobienia/PART4LAB1.2/PART4LAB1.2.ino

```diff
@@ -0,0 +1,160 @@
+int currentMode = OFF;
+
+// variables to handle button state
+int lastButtonState = HIGH;    // variable to store the last button state
+int buttonState;               // variable to store the current button state
+
+// setup function runs once at the beginning
+void setup() {
+  // set pin modes for RGB and button
+  pinMode(redPin, OUTPUT);     // set red led pin as OUTPUT
+  pinMode(greenPin, OUTPUT);   // set green led pin as OUTPUT
+  pinMode(bluePin, OUTPUT);    // set blue led pin as OUTPUT
+  pinMode(buttonPin, INPUT_PULLUP);  // set button pin as INPUT_PULLUP
+
+  // initialize serial communication for debugging
+  Serial.begin(9600);
+}
+
+// main loop function runs repeatedly
+void loop() {
+  // Read the state of the button
+  buttonState = digitalRead(buttonPin);
+
+  // check if the button is pressed (LOW) and was not pressed in the last loop
+  if (buttonState == LOW && lastButtonState == HIGH) {
+    // change to the next mode
+    changeMode();
+    // add a delay to debounce the button press
+    delay(1000);
+  }
+
+  // execute the functionality of the current mode
+  executeMode();
+
+  // save the current button state for the next loop iteration
+  lastButtonState = buttonState;
+}
+
+// function to change the mode
+void changeMode() {
+  // increment the current mode, and wrap around to 1 if it exceeds 5
+  currentMode = (currentMode % 5) + 1;
+  // print the current mode for debugging
+  Serial.print("Switching to mode ");
+  Serial.println(currentMode);
+}
+
+// function to execute the functionality of the current mode
+void executeMode() {
+  // switch statement to determine the action for each mode
+  switch (currentMode) {
+    //mode one: off
+    case OFF:
+      analogWrite(redPin, 0);
+      analogWrite(greenPin, 0);
+      analogWrite(bluePin, 0);
+      break; //breaks into void defined statements
+
+    // mode two: blink white 
+    case BLINK_WHITE:
+      blinkWhite();
+      break;
+
+    // mode three: fade purple
+    case FADE_PURPLE:
+      fadePurple();
+      break;
+
+    // mode four: fade red blue
+    case FADE_RED_BLUE:
+      fadeRedBlue();
+      break;
+
+    //mode five: flash blue and green, go seahawks!
+    case FLASH_BLUE_GREEN:
+      flashBlueGreen();
+      break;
+  }
+}
+
+
+void blinkWhite() {
+  // set the red, green and blue pins to maximum brightness 
+  analogWrite(redPin, 255);
+  analogWrite(greenPin, 255);
+  analogWrite(bluePin, 255);
+  // wait for 500 milliseconds
+  delay(500);
+  // turn off the red, green and blue pins
+  analogWrite(redPin, 0);
+  analogWrite(greenPin, 0);
+  analogWrite(bluePin, 0);
+  delay(500);
+}
+
+// function to fade purple from bright to dim and back
+void fadePurple() {
+  // loop from 0 to 255 with increments of 5
+  for (int brightness = 0; brightness <= 255; brightness += 5) {
+    // set the red and blue pins to the same brightness value
+    analogWrite(redPin, brightness);
+    analogWrite(bluePin, brightness);
+    delay(50);
```

Comment:
This delay gets repeated 255/5 times and repeated below, all before the button is checked again. That's 5.1 seconds before the button is checked again.